src/index.c

```diff
@@ -552,7 +552,7 @@ int git_index_clear(git_index *index)
 
 static int create_index_error(int error, const char *msg)
 {
-	giterr_set_str(GITERR_INDEX, msg);
+	giterr_set_str(GITERR_INDEX, "index creation error '%s'", msg);
```

Comment:
You're changing the behavior rather drastically.  The name `create_index_error` indicates that this function _creates_ an _index error_; it doesn't indicate that there was an error _creating the index_.

This will produce errors like:

    index creation error 'Failed to read index: The index is in-memory only'

Which does not make sense.

There are a variety of options for fixing this, simply `giterr_set_str(GITERR_INDEX, "%s", msg);` is certainly the simplest.